Resources/doc/first_steps.rst

```diff
@@ -249,6 +249,226 @@ to easily fetch objects based on multiple conditions:
+    {
+        #[Route('/product/{id}')]
+        public function show(
+            #[MapDocument]
+            Product $product
+        ): Response {
+            // use the Product!
+            // ...
+        }
+    }
+
+.. tip::
+
+    When enabled globally, it's possible to disable the behavior on a specific
+    controller, by using the ``MapDocument`` set to ``disabled``:
+
+.. code-block:: php
+
+        public function show(
+            #[CurrentUser]
+            #[MapDocument(disabled: true)]
+            User $user
+        ): Response {
+            // User is not resolved by the DocumentValueResolver
+            // ...
+        }
+
+Fetch Automatically
+~~~~~~~~~~~~~~~~~~~
+
+If your route wildcards match properties on your document, then the resolver
+will automatically fetch them:
+
+.. code-block:: php
+
+    /**
+     * Fetch via primary key because {id} is in the route.
+     */
+    #[Route('/product/{id}')]
+    public function showByPk(Post $post): Response
+    {
+    }
+
+    /**
+     * Perform a findOneBy() where the slug property matches {slug}.
+     */
+    #[Route('/product/{slug}')]
+    public function showBySlug(Post $post): Response
+    {
+    }
+
+Automatic fetching works in these situations:
+
+* If ``{id}`` is in your route, then this is used to fetch by
+  primary key via the ``find()`` method.
+
+* The resolver will attempt to do a ``findOneBy()`` fetch by using
+  *all* of the wildcards in your route that are actually properties
+  on your document (non-properties are ignored).
+
+You can control this behavior by actually *adding* the ``MapDocument``
+attribute and using the `MapDocument options`_.
+
+Fetch via an Expression
+~~~~~~~~~~~~~~~~~~~~~~~
+
+If automatic fetching doesn't work, you can write an expression using the
+`ExpressionLanguage component`_:
+
+.. code-block:: php
+
+    #[Route('/product/{product_id}')]
+    public function show(
+        #[MapDocument(expr: 'repository.find(product_id)')]
+        Product $product
+    ): Response {
+    }
+
+In the expression, the ``repository`` variable will be your document's
+Repository class and any route wildcards - like ``{product_id}`` are
+available as variables.
+
+This can also be used to help resolve multiple arguments:
+
+.. code-block:: php
+
+    #[Route('/product/{id}/comments/{comment_id}')]
+    public function show(
+        Product $product,
+        #[MapDocument(expr: 'repository.find(comment_id)')]
+        Comment $comment
+    ): Response {
+    }
+
+In the example above, the ``$product`` argument is handled automatically,
+but ``$comment`` is configured with the attribute since they cannot both follow
+the default convention.
+
+MapDocument Options
+~~~~~~~~~~~~~~~~~~~
+
+A number of options are available on the ``MapDocument`` attribute to
+control behavior:
+
+``id``
+    If an ``id`` option is configured and matches a route parameter, then
+    the resolver will find by the primary key:
```

Comment:
```suggestion
    the resolver will find by the identifier:
```

I've seen `primary key` in previous chapter as well, best to check for it globally